test/test_sklearn_compat.py

```diff
@@ -105,6 +105,70 @@ def stable_init(self, n_components=None, pca_comps=None,
 
 # ---------------------- Test scikit-learn compatibility ----------------------
 
+def generate_array_like(input_data, labels=None):
+  """Helper function to generate array-like variants of numpy datasets,
+  for testing purposes."""
+  list_data = input_data.tolist()
+  input_data_changed = [input_data, list_data, tuple(list_data)]
+  if input_data.ndim >= 2:
+    input_data_changed.append(tuple(tuple(x) for x in list_data))
+  if input_data.ndim >= 3:
+    input_data_changed.append(tuple(tuple(tuple(x) for x in y) for y in
+                                    list_data))
+  if input_data.ndim == 2:
+    pd = pytest.importorskip('pandas')
+    input_data_changed.append(pd.DataFrame(input_data))
+  if labels is not None:
+    labels_changed = [labels, list(labels), tuple(labels)]
+  else:
+    labels_changed = [labels]
+  return input_data_changed, labels_changed
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize('with_preprocessor', [True, False])
+@pytest.mark.parametrize('estimator, build_dataset', metric_learners,
+                         ids=ids_metric_learners)
+def test_array_like_inputs(estimator, build_dataset, with_preprocessor):
+  """Test that metric-learners can have as input (of all functions that are
+  applied on data) any array-like object."""
+  input_data, labels, preprocessor, X = build_dataset(with_preprocessor)
+
+  # we subsample the data for the test to be more efficient
+  input_data, _, labels, _ = train_test_split(input_data, labels,
+                                              train_size=20)
```

Comment:
maybe setting `stratify=labels` would make LMNN pass for `train_size=10`?

Reply:
You're right, I tried and it worked for 11 (though it didn't work for 10, I think because the choice of chunks removes any point once it's already included in a chunk or something like that, and the dataset has 3 classes, but I would need to investigate more to be sure) 
However, it bugs for MLKR, since labels is a continuous values array that cannot be used as an argument of "stratify"
So I suggest that to keep it simple we keep it like that (with 20 samples, and no stratify), what do you think ?

Reply:
Fair enough, let's merge